### PR TITLE
Tpt3 76

### PR DESCRIPTION
**Pull request by zemlyanoi**

### CODE REVIEW

File: src/main/java/gov/ca/cwds/cm/web/rest/ResponseUtil.java [hunk 1]
```diff
@@ -14,35 +14,42 @@
...
         .entity(isCollectionEmpty ? null : collection)
         .build();
   }
+
+  public static Response responseOrNotFound(final gov.ca.cwds.rest.api.Response[] response) {
+    boolean isResponseNull = response == null ? true : false;
+    boolean isCollectionEmpty = false;
+    if (!isResponseNull) {
+      isCollectionEmpty = response.length == 0 ? true : false;
+    }
```
Comment (denys-davydov) [inline src/main/java/gov/ca/cwds/cm/web/rest/ResponseUtil.java]: Lines [46-50] can be replaced with 1 line:
boolean isCollectionEmpty = response == null || response.length == 0;

File: src/main/java/gov/ca/cwds/cm/web/rest/ResponseUtil.java [hunk 2]
```diff
@@ -14,35 +14,42 @@
...
+      final Collection<? extends gov.ca.cwds.rest.api.Response> collection) {
     final boolean isCollectionEmpty = CollectionUtils.isEmpty(collection);
-    return Response
-        .status(isCollectionEmpty ? HTTP_CODE_NOT_FOUND : HTTP_CODE_OK)
+    return Response.status(isCollectionEmpty ? HTTP_CODE_NOT_FOUND : HTTP_CODE_OK)
         .entity(isCollectionEmpty ? null : collection)
         .build();
   }
+
+  public static Response responseOrNotFound(final gov.ca.cwds.rest.api.Response[] response) {
```
Comment (denys-davydov) [inline src/main/java/gov/ca/cwds/cm/web/rest/ResponseUtil.java]: Please add tests for this method

File: src/main/java/gov/ca/cwds/cm/web/rest/client/ClientResource.java [hunk 3]
```diff
@@ -79,22 +77,23 @@ public Response get(
       @ApiResponse(code = 406, message = "Accept Header not supported")
     }
   )
-  @ApiOperation(value = "Find client by client ID", response = ClientDTO[].class)
+  @ApiOperation(value = "Search clients by criteria", response = ClientDTO[].class)
```
Comment (denys-davydov) [inline src/main/java/gov/ca/cwds/cm/web/rest/client/ClientResource.java]: Search operation must return ClientSearchResultDTO

**Comment by dmytro-palczewski-osi** [comment]
Should not ReferralSearchCriteria be serialized?

**Comment by dmytro-palczewski-osi** [comment]
Do we really need @AllArgsConstructor for DTOs? Probably not. But it seems that such constructors conflict with Guice injectors.

**Pull request closed without merging**